website/src/theme/DocPage/index.js

```diff
@@ -50,10 +50,31 @@ function DocPageContent({
 
   // Check if page available for current version
   const { versionedPages } = usePluginData('docusaurus-build-global-data-plugin');
-  const { version: dbtVersion, EOLDate, latestStableRelease } = useContext(VersionContext)
+  const { version: dbtVersion, EOLDate, isPrerelease, latestStableRelease } = useContext(VersionContext)
   const { pageAvailable, firstAvailableVersion } = pageVersionCheck(dbtVersion, versionedPages, currentDocRoute.path)
 
-  // Check End of Life date and show unsupported banner if depricated version
+  // Check whether this version is a isPrerelease, and show banner if so
+  const [PreData, setPreData] = useState({
+    showisPrereleaseBanner: false,
+    isPrereleaseBannerText: ''
+  })
+
+  useEffect(() => {
+    // If version is not isPrerelease, do not show banner
+    if(!isPrerelease) {
+      setPreData({
+        showisPrereleaseBanner: false,
+        isPrereleaseBannerText: ''
+      })
+    } else {
+        setPreData({
+          showisPrereleaseBanner: true,
+          isPrereleaseBannerText: `This is a prerelease version. The latest stable version is ${latestStableRelease}`
+        })
+    }
+  })
```

Comment:
@JKarlavige Ah, so that's what `}, [dbtVersion])` is doing! There's a reason you're the one who does this stuff for real ;)

Thanks for the quick pointer. I like option 2, since that keeps the logic a bit more consolidated. We shouldn't ever have a version that's _both_ a prerelease _and_ within 3 months of EOL.